tkg/client/upgrade_cluster.go

```diff
@@ -502,6 +504,20 @@ func (c *TkgClient) getUpgradeClusterConfig(options *UpgradeClusterOptions) (*Cl
 		upgradeInfo.UpgradeComponentInfo.OsInfo = azureVMImage.OSInfo
 	}
 
+	// get kube-vip image from bom
+	if com, ok := bomConfiguration.Components["kube-vip"]; ok {
+		if len(com) >= 1 {
+			if img, ok := com[0].Images["kubeVipImage"]; ok {
+				upgradeInfo.UpgradeComponentInfo.KubeVipTag = img.Tag
+				upgradeInfo.UpgradeComponentInfo.KubeVipFullImagePath = bomConfiguration.ImageConfig.ImageRepository + "/" + img.ImagePath
```

Comment:
Sorry that I am not very familiar with airgap env. Is this enough to cover the airgap case where the image repository is private?

Reply:
+1. Is TKG_CUSTOM_IMAGE_REPOSITORY case handled?

Reply:
I checked with @imikushin, this will work. Also coredns and pause also use this path

Reply:
https://github.com/vmware-tanzu/tanzu-framework/blob/main/tkg/client/upgrade_cluster.go#L118-L119
Code here

Reply:
@DanielXiao @12345lcr  Ok I checked the code https://github.com/vmware-tanzu/tanzu-framework/blob/main/tkg/tkgconfigbom/bom.go#L224
It will be overwritten with customRepository